vpr/src/route/route_timing.cpp

```diff
@@ -1982,3 +1998,20 @@ static void init_net_delay_from_lookahead(const RouterLookahead& router_lookahea
         }
     }
 }
+
+//updates router iteration information and checks for router iteration and net id breakpoints
+//stops after the specified router iteration or net id is encountered
+void update_router_info_and_check_bp(bp_router_type type, int net_id) {
+    t_draw_state* draw_state = get_draw_state_vars();
+    if (draw_state->list_of_breakpoints.size() != 0) {
+        if (type == BP_ROUTE_ITER)
+            get_bp_state_globals()->get_glob_breakpoint_state()->router_iter++;
+        else if (type == BP_NET_ID)
+            get_bp_state_globals()->get_glob_breakpoint_state()->net_id = net_id;
+        f_router_debug = check_for_breakpoints(false);
```

Comment:
same comment as f_placer_debug:  is f_router_debug global and does it need to be?

Reply:
f_router_debug is used for more than just breakpoints and was there before I started editing the code.

Reply:
OK, I think I understand it better now.  It needs a comment as it's an important variable (anything file scope or global should be important and commented).  You should comment that if this variable is true, the placer will stop and let the user examine the state of the placement in the GUI.

Reply:
resolved.